test/e2e/util.go

```diff
@@ -432,10 +433,13 @@ func waitForRCPodOnNode(c *client.Client, ns, rcName, node string) (*api.Pod, er
 		for _, pod := range pods.Items {
 			if pod.Spec.NodeName == node {
 				Logf("Pod %s found on node %s", pod.Name, node)
-				p = &pod
-				return true, nil
 			}
 		}
+		for _, pod := range pods.Items {
```

Comment:
Why did you add this loop? It appears that it will return the first pod instead of the pod on the requested node?